### PR TITLE
Add polymer lint

### PR DESCRIPTION
**Pull request by stramel**
Adjusted the Travis job for linting and added the polymer-1 rule set

### CODE REVIEW

**Comment by abdonrd** [comment]
Oh! The `polymer lint` exited with 1 if has warnings. :/

**Comment by stramel** [comment]
Yeah, should we add polymer lint to the npm lint script or should we run the npm lint script along side polymer lint in the before_script section?

**Comment by abdonrd** [comment]
@stramel yes! I have a WIP PR with that!

**Comment by abdonrd** [comment]
@stramel done! https://github.com/PolymerElements/polymer-starter-kit/pull/1012

I think it will be easy to add it over. 😁

**Comment by stramel** [comment]
Closing in favor of #1012

**Comment by abdonrd** [comment]
Noo! Haha We need your PR for polymer lint too. 😄

**Comment by stramel** [comment]
😅 Whoops! Okay, reopening.

**Comment by stramel** [comment]
@abdonrd I rebased this onto your branch to minimize diff and get your changes.

**Comment by stramel** [comment]
Opened issue Polymer/polymer-cli#699 for allowing warnings to not exit 1

**Comment by stramel** [comment]
@abdonrd Closed until Polymer/polymer-cli#699 is fixed?

**Comment by abdonrd** [comment]
Oops! Sorry! I didn't want to close this.
I guess it will be closed by itself when I merged #1012.

File: .travis.yml [hunk 1]
```diff
@@ -12,8 +12,8 @@ node_js:
   - "node"
   - "6"
 install:
-  - npm install
   - npm install -g polymer-cli
+  - npm install
```
Comment (abdonrd) [inline .travis.yml]: Why this? We can keep it as it was.

File: package.json [hunk 2]
```diff
@@ -7,8 +7,9 @@
     "eslint-plugin-html": "^2.0.0"
   },
   "scripts": {
-    "lint": "npm run lint:javascript",
+    "lint": "npm run lint:javascript && npm run lint:polymer",
```
Comment (abdonrd) [inline package.json]: What do you think about keeping it in a simple way?

```json
"lint": "npm run lint:javascript && polymer lint"
```

File: .travis.yml (hunk 1 above)
Comment (stramel) [inline .travis.yml]: Maybe it is personal preference but I like to install globals first. I can revert that

File: package.json (hunk 2 above)
Comment (stramel) [inline package.json]: I can switch to keeping it simple, just thought it would keep it more compartmentalized when/if we start adding additional linters.

**Comment by stramel** [comment]
@abdonrd Did you have any thoughts on my replies?

File: .travis.yml (hunk 1 above)
Comment (abdonrd) [inline .travis.yml]: Ok!

File: package.json (hunk 2 above)
Comment (abdonrd) [inline package.json]: For now, I prefer keeping it simple.

If the `polymer lint` need flags or we add more linters in the future, we can iterate.

Comment (stramel) [inline package.json]: Will change this then :)

**Comment by stramel** [comment]
Pending these changes, https://github.com/Polymer/polymer-cli/pull/709

File: .travis.yml (hunk 1 above)
Comment (abdonrd) [inline .travis.yml]: Sorry! I said "Ok!" for your preference haha
But it doesn't matter, both are valid. 🙂

**Comment by stramel** [comment]
Adding to `2.0-preview` may be good to just do it there since there aren't any linter warnings/issues on that branch? #1018 WDYT @abdonrd, close this one, in favor of the 2.0-preview?

**Comment by abdonrd** [comment]
I think we can keep the PR.

**Comment by stramel** [comment]
@abdonrd Seems like this would be some duplicated efforts since you guys have already tagged a v3 of PSK which will have these changes and this will be outstanding until there is a decision on how to handle minSeverity. I'm going to close this for now.

We can open it back up if there is a resolution with minSeverity, or we decide to backport fixes for the lint issues and release an update to a previous version.

**Pull request closed without merging**